ValkeyFullTextSearch.md

```diff
@@ -0,0 +1,392 @@
+
+```
+## Specification
+
+Each ```TEXT``` field has the a set of configurables some control the process to convert a string into a vector of terms, others control the contents of the generated index.
+
+| Metadata          | Type              | Meaning                                                                    | Default Value |
+| ----------------- | ----------------- | -------------------------------------------------------------------------- | ------------- |
+| Punctuation       | vector<character> | Characters that separate words during the tokenization process.            |
+| Stop Words        | vector<String>    | List of stop words to be removed during decomposition                      |
+| Stemming Language | Enumeration       | Snowball stemming library language control                                 |
+| Suffix Tree       | Boolean           | Controls the presence/absence of a suffix tree.                            |
+| Word Offsets      | Boolean           | Indicates if the postings for a word contain word offsets or just a count. |
+
+### Extension of the ```FT.CREATE``` command
+
+Clauses are provided to control the configuration. 
+If supplied before the ```SCHEMA``` keyword then the default value of the clause is changed for any text fields declared in this command.
+If supplied as part of an individual field declaration, i.e., after the ```SCHEMA``` keyword, then it sets the configurable for just that field.
+
+```
+[PUNCTUATION <string>]
+```
+The characters of this string are used to split the input string into words. Note, the splitting process allows escaping of input characters using the usual backslash notation. This string cannot be empty. Default value is: _TBD_
+
+
+```
+[WITHSUFFIXTRIE | NOSUFFIXTRIE]
+```
+
+Controls the presence/absence of the second prefix tree in the field-index. Default is ```NOSUFFIXTRIE```.
+
+```
+[WITHOFFSETS | NOOFFSETS]
+```
+
+Controls whether term-offsets are tracked in the field-index. Default is ```WITHOFFSETS```.
+
+```
+[NOSTOPWORDS | STOPWORDS <count> [word ...]]
+```
+
+Controls the application and list of stop words. Note that ```STOPWORDS 0``` is equivalent to ```NOSTOPWORDS```. The default stop words are:
+
+**a,    is,    the,   an,   and,  are, as,  at,   be,   but,  by,   for,
+ if,   in,    into,  it,   no,   not, of,  on,   or,   such, that, their,
+ then, there, these, they, this, to,  was, will, with**.
+
+```
+[MINSTEMSIZE <size>]
+```
+
+This clause controls the minimum number of characters in a word before it is subjected to stemming. Default value is 4.
+
+```
+[NOSTEM | LANGUAGE <language>]
+```
+
+Controls the stemming algorithm. Supported languages are: 
+
+**none, arabic, armenian, basque, catalan, danish, dutch, english, estonian, finnish, french, german, greek, hindi, hungarian, indonesian, irish, italian, lithuanian, nepali, norwegian, porter, portuguese, romanian, russian, scripts, serbian, spanish, swedish, tamil, turkish, yiddish**
```

Comment:
miss: hebrew. Consider something pluggable in cases where Snowball doesn't cover.